src/main/java/com/Model/Patient.java

```diff
@@ -13,25 +13,28 @@
 public class Patient {
 
     @Id
-    private String Id;
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long Id;
+
+    private String sessionId;
 
     @OneToMany (cascade = CascadeType.ALL)
     private Set <PatientEventAssociation> patientEventAssociations;
 
     //All patients are anonymous, name is not needed for constructor
     @Autowired
     public Patient(){
-        this.Id = UUID.randomUUID().toString();
+        this.sessionId = UUID.randomUUID().toString();
```

Comment:
New idea: I don't think we actually need a Session id. We can just use the URI of the patient resource that we're returning to the client upon creation. The client would need to submit the Uri when adding the event.